pkg/build/builder/sti.go

```diff
@@ -92,9 +92,16 @@ func (s *S2IBuilder) Build() error {
 	if s.build.Spec.Strategy.SourceStrategy == nil {
 		return fmt.Errorf("the source to image builder must be used with the source strategy")
 	}
-
+	var runtimeImageName string
 	var push bool
 
+	runtimeImage := s.build.Spec.Strategy.SourceStrategy.RuntimeImage
```

Comment:
is there any particular reason why, for example the From ref in sourcebuildstrategy is not a pointer yet DockerBuildStrategy is https://github.com/openshift/origin/blob/master/pkg/build/api/types.go#L475 and then, again Custom isn't  https://github.com/openshift/origin/blob/master/pkg/build/api/types.go#L444
are we concerned with mutating it?

should i not use a pointer after all for runtimeImage? https://github.com/openshift/origin/pull/9596/files#diff-f23c2970936ec5cfcb60d485ae96bcffR519